### ISSUE TITLE
SaveSpatialMapping Demo Fails to load correct Transforms

### ISSUE THREAD

**Issue by AdamRVierra** [Microsoft/MixedRealityToolkit-Unity#817]
Unity Version: 5.6.1f1. 

Steps:
1: Play the SaveSpatialMapping scene.
2: Wait for room to scan. Once enough is scanned, save. Observe the parts of your room in the inspector. It has unique value for the position and rotation.

![spatialmapping1](https://user-images.githubusercontent.com/10157808/28799128-25a1b122-75fc-11e7-8af5-cfcccabc2df0.png)

3: Restart program and load the room you saved earlier.
4: Mesh geometry is properly loaded in. However, every part of your previously scanned room is loaded in with the local position and local rotation of 0,0,0.

![spatialmapping2](https://user-images.githubusercontent.com/10157808/28799129-25a1e6ba-75fc-11e7-911a-c4dceb937eef.png)

Tested in real space (deployed as a Hololens Project) and Hololens Emulator (Every sample room) and multiple previous Unity versions and multiple computers.

**Comment by StephenHodgson** [comment]
Thanks for the report.

**Comment by AdamRVierra** [comment]
I was able to mostly fix this issue on my end.

While saving each mesh piece, I attached a World Anchor to it and saved it. I named the anchor the same name as the mesh piece. I had to save the mesh names in an xml file because the mesh pieces don't retain their names when being loaded from the .room file. I load each piece one by one and attach an anchor to the piece, going through the name list in order.

There may be some risk involved, since it assumes that every mesh is saved and loaded in the exact same order, but I did a fair amount of testing and haven't noticed anything odd. There's also several xml methods that are unavailable to Hololens, but there's enough to just save and load the list.

I'm sure there's better solutions, but I hope this at least points people in the right direction for now.
